client/src/unifyfs-fixed.c

```diff
@@ -397,83 +553,53 @@ static int unifyfs_logio_chunk_write(
     unifyfs_index_t cur_idx;
     cur_idx.fid      = ptr_meta_entry->gfid;
     cur_idx.file_pos = pos;
-    cur_idx.mem_pos  = log_offset;
+    cur_idx.log_pos  = log_offset;
     cur_idx.length   = count;
 
-    /* split the write requests larger than unifyfs_key_slice_range into
-     * the ones smaller than unifyfs_key_slice_range */
-    index_set_t tmp_index_set;
-    memset(&tmp_index_set, 0, sizeof(tmp_index_set));
-    unifyfs_split_index(&cur_idx, &tmp_index_set,
-                        unifyfs_key_slice_range);
-
     /* lookup number of existing index entries */
     off_t num_entries = *(unifyfs_indices.ptr_num_entries);
 
-    /* number of new entries we may add */
-    off_t tmp_entries = (off_t) tmp_index_set.count;
-
-    /* check whether there is room to add new entries */
-    if (num_entries + tmp_entries < unifyfs_max_index_entries) {
-        /* get pointer to index array */
-        unifyfs_index_t* idxs = unifyfs_indices.index_entry;
-
-        /* coalesce contiguous indices */
-        int i = 0;
-        if (num_entries > 0) {
-            /* pointer to last element in index array */
-            unifyfs_index_t* prev_idx = &idxs[num_entries - 1];
-
-            /* pointer to first element in temp list */
-            unifyfs_index_t* next_idx = &tmp_index_set.idxes[0];
-
-            /* offset of last byte for last index in list */
-            off_t prev_offset = prev_idx->file_pos + prev_idx->length;
-
-            /* check whether last index and temp index refer to
-             * contiguous bytes in the same file */
-            if (prev_idx->fid == next_idx->fid &&
-                    prev_offset   == next_idx->file_pos) {
-                /* got contiguous bytes in the same file,
-                 * check if both index values fall in the same slice */
-                off_t prev_slice = prev_idx->file_pos / unifyfs_key_slice_range;
-                off_t next_slice = next_idx->file_pos / unifyfs_key_slice_range;
-                if (prev_slice == next_slice) {
-                    /* index values also are in same slice,
-                     * so append first index in temp list to
-                     * last index in list */
-                    prev_idx->length  += next_idx->length;
-
-                    /* advance to next index in temp list */
-                    i++;
-                }
-            }
-        }
+    /* remaining entries we can fit in the shared memory region */
+    off_t remaining_entries = unifyfs_max_index_entries - num_entries;
```

Comment:
Re: prior comment. Should bail out with error here if `remaining_entries == 0`.

Reply:
There is one corner case in which the write index can be fully coalesced with the last index, so that it doesn't require adding an additional entry.  We need to throw an error if ```remaining_entries==0``` and we need to call the split routine, though.  Currently that logic is inside of the split function, but I'll pull it out here.

Reply:
ok, sounds good